clients/gateway_test.go

```diff
@@ -517,11 +517,15 @@ func TestGetClassDefinition(t *testing.T) {
 }
 
 func TestHttpError(t *testing.T) {
+	maxRetries := 2
+	callCount := make(map[string]int)
 	srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		soFar := callCount[r.URL.String()]
+		callCount[r.URL.String()] = soFar + 1
```

Comment:
```suggestion
		callCount[r.URL.String()]++
```

Not entirely sure the above will work but worth a try.

Reply:
works:) thanks!